### DISCUSSION TITLE
[2023-03-20] Library and Engine Team Update

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#6016] [Weekly Demo Day]
- James: Work on integrating ag-grid in place of JS table viz. #5915 
- James: Design requirement for data grid view longer term. #5965 
- James: Design for execution modes and the play button. #5964 (awaiting review)
- James: Aligned type signatures for method. #5979
- Dmitry: Fixed tagValues for constructors so correct FQN. #5978 
- James: Resolved some issues with drop downs on methods with warnings attached. #5994 (passed remaining issue to Hubert as low level in engine)
- Radek: Integrated new Value Type into the in-memory storage types and repaired all the failing tests.
- Radek: Working on deriving Value Type from the results of JDBC queries.
- Greg: Aligned the various accessor methods for groups on the Match_2 type. #5883 
- Greg: Working on the replace and replace all functions. #5959 
- Jaroslav: `Meta.get_constructor_fields` works for a `Type`/ #5869 
- Jaroslav: Defer serialization of IR. 
- Jaroslav: Avoid UUIDs in IR files. #5918 
- Jaroslav: Adding `Meta.Type`. #5956 
- Jaroslav: Fix for `UpsertVisualizationJob` performance. #5945 
- Hubert: Avoid GraalVM installing component multiple times. #5900
- Hubert: Resolving issue with exports causing conflicts with FQNs. #5902 
- Hubert: Adding indexing to buildStdLib. #5999 
- Hubert: Looking at the current execution context code.
- Hubert: Investigating issue in instrumentation such that callMethod is unset.
- Pavel M: Finished the benchmark analysis tool. #5852 
- Pavel M: Implement Enso-specific asserts. #5973
- Pavel M: Sort handles Incomparable types by default. #5998 
- Dmitry: Spurious visualization request timeouts. #5913 
- Dmitry: Writing down IR caches slows IDE startup. #5957
- Dmitry: Parallelize the SQL DB initialization. #5970 
- Dmitry: Parallelize the JSON protocol init. #5997 
- Pawel G: Drop down display labels from `Single_Choice`. #5705 
- Pawel G: Simplify Node Edit mode.
- Pawel G: Planning and tasking for widgets on nodes.

**Comment by radeusgd** [comment]
Progress update on Value_Types - we have type mappings between `Value_Type` and SQL types for SQLite and Postgres along with unit tests, although the mappings require some further care to be considered finished.

Instead of implementing our own logic for inferring result types which may easily get out of sync if Database versions differ, we now rely on the Database itself to tell us the result type:

(not doing a screenshot as my build is currently logging tons of debug information that is just noise for presenting)
```
> c2.execute_update 'CREATE TABLE t123 (x int2, y int4, z int8, w float);'
> t123 = c2.query (SQL_Query.Table_Name "t123")
> t123.info.print
   | Column | Items Count | Value Type       
---+--------+-------------+-------------------
 0 | x      | 0           | (Integer 16 bits)
 1 | y      | 0           | (Integer 32 bits)
 2 | z      | 0           | (Integer 64 bits)
 3 | w      | 0           | (Float 64 bits)

> (t123.at 'x' + t123.at 'y') . value_type
>>> (Integer 32 bits)

> (t123.at 'x' + t123.at 'z') . value_type
>>> (Integer 64 bits)

> (t123.at 'x' + t123.at 'w') . value_type
>>> (Float 64 bits)

> t123.aggregate [Aggregate_Column.Group_By 'z', Aggregate_Column.Sum 'x', Aggregate_Column.Maximum 'x'] . info . print
   | Column    | Items Count | Value Type       
---+-----------+-------------+-------------------
 0 | z         | 0           | (Integer 64 bits)
 1 | Sum x     | 0           | (Integer 64 bits)
 2 | Maximum x | 0           | (Integer 16 bits)
```
